great_expectations/execution_engine/execution_engine.py

```diff
@@ -327,7 +343,7 @@ def resolve_metrics(
             for k, v in metric_to_resolve.metric_dependencies.items():
                 if v.id in metrics:
                     metric_dependencies[k] = metrics[v.id]
-                elif self._caching and v.id in self._metric_cache:
+                elif self._caching and v.id in self._metric_cache:  # type: ignore[operator] # TODO: update NoOpDict
```

Comment:
`NoOpDict` needs the `__contains__` operator implemented, to deal with this issue.

The question I have here... is whether should it always return `False` or always return `True`? 
I think always returning `True` is a lot more dangerous.